src/nvim/highlight_defs.h

```diff
@@ -0,0 +1,126 @@
+#ifndef NVIM_HIGHLIGHT_DEFS_H
+#define NVIM_HIGHLIGHT_DEFS_H
```

Comment:
I started working on it before (#1933), but stopped as it was decided that syntax.c was "frozen". But we already have several separate _defs files even if the .c is not.